src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -320,6 +320,19 @@ namespace winrt::TerminalApp::implementation
         if (_startupState == StartupState::NotInitialized)
         {
             _startupState = StartupState::InStartup;
+
+            // If the user selected to save their tab layout, we are the first
+            // window opened, and wt was not run with any other arguments, then
+            // we should use the saved settings.
+            if (_settings.GlobalSettings().PersistTabLayout() && _WindowId == 1 && _startupActions.Size() == 1)
```

Comment:
Open to other ideas on how we should handle multiple windows. (See my thoughts in the pull description)